aws_requests_auth/boto_utils.py

```diff
@@ -4,20 +4,50 @@
 aws-requests-auth package.
 """
 
-import boto3
+from botocore.session import Session
 
+from .aws_auth import AWSRequestsAuth
 
-def get_credentials():
+
+def get_credentials(credentials_obj=None):
     """
     Interacts with boto to retrieve AWS credentials, and returns a dictionary of
-    kwargs to be used in AWSRequestsAuth. boto automatically pulls AWS credentials from 
+    kwargs to be used in AWSRequestsAuth. boto automatically pulls AWS credentials from
     a variety of sources including but not limited to credentials files and IAM role.
-    AWS credentials are pulled in the order listed here: 
+    AWS credentials are pulled in the order listed here:
     http://boto3.readthedocs.io/en/latest/guide/configuration.html#configuring-credentials
     """
-    credentials_obj = boto3.Session().get_credentials()
+    if credentials_obj is None:
+        credentials_obj = Session().get_credentials()
+    # use get_frozen_credentials to avoid the race condition where one or more
+    # properties may be refreshed and the other(s) not refreshed
+    frozen_credentials = credentials_obj.get_frozen_credentials()
     return {
-        'aws_access_key': credentials_obj.access_key,
-        'aws_secret_access_key': credentials_obj.secret_key,
-        'aws_token': credentials_obj.token,
+        'aws_access_key': frozen_credentials.access_key,
+        'aws_secret_access_key': frozen_credentials.secret_key,
+        'aws_token': frozen_credentials.token,
     }
+
+
+class BotoAWSRequestsAuth(AWSRequestsAuth):
+
+    def __init__(self, aws_host, aws_region, aws_service):
+        """
+        Example usage for talking to an AWS Elasticsearch Service:
+
+        BotoAWSRequestsAuth(aws_host='search-service-foobar.us-east-1.es.amazonaws.com',
+                            aws_region='us-east-1',
+                            aws_service='es')
+
+        The aws_access_key, aws_secret_access_key, and aws_token are discovered
+        automatically from the environment, in the order described here:
+        http://boto3.readthedocs.io/en/latest/guide/configuration.html#configuring-credentials
+        """
+        super().__init__(None, None, aws_host, aws_region, aws_service)
```

Comment:
is this call to `super` compatible with python 2?

I believe python 2 will crash on a call to super with 0 args:

```python
In [10]: import sys

In [11]: sys.version
Out[11]: '2.7.6 (default, Mar 22 2014, 22:59:56) \n[GCC 4.8.2]'

In [12]: super?
Docstring:
super(type, obj) -> bound super object; requires isinstance(obj, type)
super(type) -> unbound super object
super(type, type2) -> bound super object; requires issubclass(type2, type)
Typical use to call a cooperative superclass method:
class C(B):
    def meth(self, arg):
        super(C, self).meth(arg)
Type:      type

In [13]: super()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-13-d008e6bbc038> in <module>()
----> 1 super()

TypeError: super() takes at least 1 argument (0 given)

In [14]: 
```